File: 3066-minimum-operations-to-exceed-threshold-value-ii/main.py
class Solution:
    def minOperations(self, nums: List[int], k: int) -> int:
        
        def _helper(nums: List[int], n_operations: int) -> int:
            
            heapq.heapify(nums)

            while nums[0] < k:

                if len(nums) < 2:
                    return n_operations
                
                smallest = heapq.heappop(nums)
                second_smallest = heapq.heappop(nums)

                new_element = smallest * 2 + second_smallest
                heapq.heappush(nums, new_element)

                n_operations += 1
            
            return n_operations

        return _helper(nums, 0)

File: 3066-minimum-operations-to-exceed-threshold-value-ii/main.go
type MinHeap []int

func (h MinHeap) Len() int { return len(h) }
func (h MinHeap) Less(i, j int) bool { return h[i] < h[j] }
func (h MinHeap) Swap(i, j int)      { h[i], h[j] = h[j], h[i] }

func (h *MinHeap) Push(x interface {}) {
    *h = append(*h, x.(int))
}
func (h *MinHeap) Pop() interface{} {
    old := *h
    n := len(old)
    x := old[n-1]
    *h = old[0: n-1]
    return x
}

func minOperations(nums []int, k int) int {
    h := MinHeap(nums)
    heap.Init(&h)

    numOperations := 0

    for h[0] < k {
        
        if len(h) < 2 {
            return numOperations
        }

        smallest := heap.Pop(&h).(int)
        secondSmallest := heap.Pop(&h).(int)

        newElement := smallest*2 + secondSmallest
        heap.Push(&h, newElement)

        numOperations++
    }

    return numOperations
}
